homeassistant/components/camera/neato.py

```diff
@@ -0,0 +1,98 @@
+"""
+Camera that loads a picture from a local file.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.neato/
+"""
+import errno
+import logging
+import os
+
+from homeassistant.components.camera import Camera
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.components.neato import (
+    NEATO_MAP_DATA, NEATO_ROBOTS, NEATO_LOGIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Camera."""
+    if not hass.data[NEATO_ROBOTS]:
+        return False
+    directory_path = hass.config.config_dir
+    if not os.access(directory_path, os.R_OK):
+        _LOGGER.error("file path %s is not readable", directory_path)
+        return False
+    dev = []
+    for robot in hass.data[NEATO_ROBOTS]:
+        if 'maps' not in robot.traits:
+            return False
+        dev.append(NeatoCleaningMap(hass, robot, directory_path))
+    _LOGGER.debug('Adding robots for cleaning maps %s', dev)
+    directory_path = hass.config.config_dir
+    add_devices(dev)
+
+
+class NeatoCleaningMap(Camera):
+    """Neato cleaning map for last clean."""
+
+    def __init__(self, hass, robot, directory_path):
+        """Initialize Neato cleaning map."""
+        super().__init__()
+        self.robot = robot
+        self._robot_name = self.robot.name + ' Cleaning Map'
+        self._robot_serial = self.robot.serial
+        self._directory_path = directory_path
```

Comment:
I'm not sure why this is necessary, since it's just the config dir of hass. Instead, since `hass` is available on the object, you can use `self.hass.config.path(self._filename)`